### PR TITLE
Implement backup/restore to Azure blob storage

### PR DESCRIPTION
**Pull request by mrbrowning**

### CODE REVIEW

File: frameworks/cassandra/src/main/dist/svc.yml [hunk 1]
```diff
@@ -103,6 +103,14 @@ pods:
         goal: FINISHED
         cmd: aws s3 cp container-path/snapshot/ s3://${S3_BUCKET_NAME}/${SNAPSHOT_NAME}/node-${POD_INSTANCE_INDEX}/ --recursive
         resource-set: sidecar-resources
+      upload-azure:
+        goal: FINISHED
+        cmd: >
+                az login -u $CLIENT_ID --service-principal --tenant $TENANT_ID -p $CLIENT_SECRET
```
Comment (mohitsoni) [inline frameworks/cassandra/src/main/dist/svc.yml]: I'm sure this is correct, but I'll still ask. Does `--service-principal` takes an argument like other flags in this command ?

Comment (mrbrowning) [inline frameworks/cassandra/src/main/dist/svc.yml]: No, it's just a boolean flag that tells the Azure CLI client what type of principal the argument to `-u` is. Seems to work as exercised by the integration test :)

**Comment by mrbrowning** [comment]
@mesosphere-ci retest this please

File: frameworks/cassandra/src/main/dist/svc.yml (hunk 1 above)
Comment (mohitsoni) [inline frameworks/cassandra/src/main/dist/svc.yml]: 👍

**Comment by mohitsoni** [comment]
@mesosphere-ci retest this please

**Comment by mrbrowning** [comment]
@mesosphere-ci retest this please

**Comment by mrbrowning** [comment]
Cluster launch is failing.

**Comment by jsrodman** [comment]
Failed in nightly, please take a look.

https://jenkins.mesosphere.com/service/jenkins/job/dcos-commons/job/infinity-dcos-commons-nightly/98/display/redirect?page=changes

**Comment by mrbrowning** [comment]
Looks like I failed to add the relevant secrets for Azure access, fixed and rerunning now.

**Pull request merged**